### PR TITLE
cloud: Create migrate-incremental-data-from-mysql.md

### PR DESCRIPTION
**Pull request by hfxsd**
### What is changed, added or deleted? (Required)

Added a new article about incremental data migration from MySQL to TiDB Cloud.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v6.2 (TiDB 6.2 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+# Task mode. The "incremental" mode only performs incremental data migration.
+task-mode: incremental
+# timezone: "UTC"
+
+## ******** Data Source Configuration **********
+## (Optional) If you need to incrementally replicate data that has already been migrated in the full data migration, you need to enable the safe mode to avoid the incremental data migration error.
+##  This scenario is common in the following case: the full migration data does not belong to the data source's consistency snapshot, and after that, DM starts to replicate incremental data from a position earlier than the full migration.
+syncers:        	# The running configurations of the sync processing unit.
+  global:       	# Configuration name.
+  safe-mode: true   # If this field is set to true, DM changes INSERT of the data source to REPLACE for the target database, and changes UPDATE of the data source to DELETE and REPLACE for the target database. This is to ensure that when the table schema contains a primary key or unique index, DML statements can be imported repeatedly. In the first minute of starting or resuming an incremental migration task, DM automatically enables the safe mode.
```
Comment (sunzhaoyang) [inline tidb-cloud/migrate-incremental-data-from-mysql.md:97]: false is suggested as default value.

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+mysql-instances:
+  - source-id: "mysql-replica-01"
+	block-allow-list:  "bw-rule-1"
+	route-rules: ["route-rule-1"]
+	filter-rules: ["tpcc-filter-rule"]
+	syncer-config-name: "global"                	# You can use the syncers incremental data configuration above.
+	meta:       	                                # When task-mode is "incremental" and the target database does not have a checkpoint, DM uses the binlog position as the starting point. If the target database has a checkpoint, DM uses the checkpoint as the starting point.
+  	binlog-name: "mysql-bin.000001"
```
Comment (sunzhaoyang) [inline tidb-cloud/migrate-incremental-data-from-mysql.md:106]: indent looks incorrect

File: tidb-cloud/migrate-incremental-data-from-mysql.md (hunk 1 above)
Comment (sunzhaoyang) [reply tidb-cloud/migrate-incremental-data-from-mysql.md:97]: And the indent looks incorrect

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+
+## ******** Function Configuration **********
+block-allow-list:
+  bw-rule-1:
+	do-dbs: ["~^tpcc.*"]
+
+routes:                       	# Table renaming rules ('routes') from upstream to downstream tables, in order to support merging different tables into a single target table.
+  route-rule-1:               	# Rule name.
+	schema-pattern: "tpcc"    	# Rule for matching upstream schema names. It supports the wildcards "*" and "?".
+	target-schema: "tpcc"     	# Name of the target schema.
```
Comment (sunzhaoyang) [inline tidb-cloud/migrate-incremental-data-from-mysql.md:125]: useless if the schema-pattern and target-schema are the same .

Comment (hfxsd) [reply tidb-cloud/migrate-incremental-data-from-mysql.md:125]: ok, changed to tpdd

File: tidb-cloud/migrate-incremental-data-from-mysql.md (hunk 2 above)
Comment (hfxsd) [reply tidb-cloud/migrate-incremental-data-from-mysql.md:106]: fixed

File: tidb-cloud/migrate-incremental-data-from-mysql.md (hunk 1 above)
Comment (hfxsd) [reply tidb-cloud/migrate-incremental-data-from-mysql.md:97]: > false is suggested as default value.

You are right. Changed to false

**Review by Frank945946 — APPROVED** [review]
LGTM

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+Started dump at: 2022-05-24 11:19:37
+SHOW MASTER STATUS:
+    Log: mysql-bin.000001
+    Pos: 77092852
+    GTID:b631bcad-bb10-11ec-9eee-fec83cf2b903:1-640
+
+Finished dump at: 2022-05-24 11:19:53
+```
+
+Based on the above starting point information, create a migration task as follows.
```
Comment (qiancai) [inline tidb-cloud/migrate-incremental-data-from-mysql.md:90]:
```suggestion
Based on the above starting point information, create a migration task as follows:
```

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+    block-allow-list:  "bw-rule-1"
+    route-rules: ["route-rule-1"]
+    filter-rules: ["tpcc-filter-rule"]
+    syncer-config-name: "global"                   # You can use the syncers incremental data configuration above.
+    meta:                                          # When task-mode is "incremental" and the target database does not have a checkpoint, DM uses the binlog position as the starting point. If the target database has a checkpoint, DM uses the checkpoint as the starting point.
+    binlog-name: "mysql-bin.000001"
+    binlog-pos: 77092852
+    binlog-gtid: "b631bcad-bb10-11ec-9eee-fec83cf2b903:1-640"
+
+## ******** Configuration of the Target TiDB on TiDB Cloud **********
```
Comment (qiancai) [inline tidb-cloud/migrate-incremental-data-from-mysql.md:118]:
```suggestion
## ******** Configuration of the target TiDB cluster on TiDB Cloud **********
```

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+target-database:    # the Target TiDB on TiDB Cloud
```
Comment (qiancai) [inline tidb-cloud/migrate-incremental-data-from-mysql.md:119]:
```suggestion
target-database:    # the target TiDB on TiDB Cloud
```

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+  host: "tidb.70593805.b973b556.ap-northeast-1.prod.aws.tidbcloud.com"
+  port: 4000
+  user: "root"
+  password: "oSWRLvR3F5GDIgm+l+9h3kB72VFWBUwzOw=="     # If the password is not empty, it is recommended to use a dmctl-encrypted cipher
```
Comment (qiancai) [inline tidb-cloud/migrate-incremental-data-from-mysql.md:123]:
```suggestion
  password: "oSWRLvR3F5GDIgm+l+9h3kB72VFWBUwzOw=="     # If the password is not empty, it is recommended to use a dmctl-encrypted cipher.
```

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 8]
```diff
@@ -0,0 +1,245 @@
...
+    events: ["drop database"]
+    action: Ignore
+
+##  ******** Ignore check items **********
+ignore-checking-items: ["table_schema"]
+```
+
+For detailed task configurations, see [DM Task Configurations](https://docs.pingcap.com/tidb/stable/task-configuration-file-full).
+
+To run a data migration task smoothly, DM triggers a precheck automatically at the start of the task and returns the check results. DM starts the migration only after the precheck is passed. To trigger a precheck manually, run the check-task command.
```
Comment (qiancai) [inline tidb-cloud/migrate-incremental-data-from-mysql.md:147]:
```suggestion
To run a data migration task smoothly, DM triggers a precheck automatically at the start of the task and returns the check results. DM starts the migration only after the precheck is passed. To trigger a precheck manually, run the `check-task` command.
```

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 9]
```diff
@@ -0,0 +1,245 @@
...
+
+```shell
+[root@localhost ~]# tiup dmctl --master-addr 172.16.7.140:9261 check-task dm-task1.yaml
+tiup is checking updates for component dmctl ...
```
Comment (qiancai) [inline tidb-cloud/migrate-incremental-data-from-mysql.md:151]: Please separate the output from the command.

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 10]
```diff
@@ -0,0 +1,245 @@
...
+    "msg": "check pass!!!"
+}
+```
+
+## Step 4. Start the migration task
+
+Run the following command to start the migration task:
+
+```shell
+[root@localhost ~]# tiup dmctl --master-addr 172.16.7.140:9261 start-task dm-task1.yaml
```
Comment (qiancai) [inline tidb-cloud/migrate-incremental-data-from-mysql.md:164]: the filename is dm-task1.yaml here but `task.yaml` in Step 3. Please use one of them consistently.

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 11]
```diff
@@ -0,0 +1,245 @@
...
+tiup is checking updates for component dmctl ...
+Starting component `dmctl`: /root/.tiup/components/dmctl/v6.0.0/dmctl/dmctl /root/.tiup/components/dmctl/v6.0.0/dmctl/dmctl --master-addr 172.16.7.140:9261 start-task dm-task1.yaml
```
Comment (qiancai) [inline tidb-cloud/migrate-incremental-data-from-mysql.md:166]: Please separate the output from the command.

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 12]
```diff
@@ -0,0 +1,245 @@
...
+    "checkResult": ""
+}
+```
+
+The parameters used in the command above are described as follows:
+
+|Parameter              |Description    |
+|-                      |-              |
+|`--master-addr`        |The {advertise-addr} of any DM-master in the cluster where `dmctl` is to be connected, e.g.: 172.16.10.71:8261|
+|`start-task`           |Starts the migration task.|
```
Comment (qiancai) [inline tidb-cloud/migrate-incremental-data-from-mysql.md:187]: Please move this part after L164

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 13]
```diff
@@ -0,0 +1,245 @@
...
+
+If you encounter any problem, refer to [DM error handling](https://docs.pingcap.com/tidb/stable/dm-error-handling) and [DM FAQ](https://docs.pingcap.com/tidb/stable/dm-faq).
+
+## Step 5. Check the migration task status
+
+To learn whether the DM cluster has an ongoing migration task and view the task status, run the `query-status` command using `tiup dmctl`:
+
+```shell
+[root@localhost ~]# tiup dmctl --master-addr 172.16.7.140:9261 query-status test-task1
+tiup is checking updates for component dmctl ...
```
Comment (qiancai) [inline tidb-cloud/migrate-incremental-data-from-mysql.md:199]: Please separate the output from the command.

**Review by qiancai — COMMENTED** [review]
@hfxsd I've completed the review and added some comments. PTAL. Thanks.

File: tidb-cloud/migrate-incremental-data-from-mysql.md (hunk 10 above)
Comment (hfxsd) [reply tidb-cloud/migrate-incremental-data-from-mysql.md:171]: OK. Changed to dm-task1.yaml

File: tidb-cloud/migrate-incremental-data-from-mysql.md (hunk 12 above)
Comment (hfxsd) [reply tidb-cloud/migrate-incremental-data-from-mysql.md:194]: Done

File: tidb-cloud/migrate-incremental-data-from-mysql.md (hunk 13 above)
Comment (hfxsd) [reply tidb-cloud/migrate-incremental-data-from-mysql.md:206]: Done

File: tidb-cloud/migrate-incremental-data-from-mysql.md (hunk 9 above)
Comment (hfxsd) [reply tidb-cloud/migrate-incremental-data-from-mysql.md:158]: Done

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 14]
```diff
@@ -0,0 +1,245 @@
...
+    ]
+}
+```
+
+The parameters used in the command above are described as follows:
+
+|Parameter              |Description    |
+|-                      |-              |
+|`--master-addr`        |The `{advertise-addr}` of any DM-master in the cluster where `dmctl` is to be connected, e.g.: 172.16.10.71:8261|
+|`operate-source create`|Loads the data source to the DM cluster.|
```
[the remark this answers is not in the document]
Comment (hfxsd) [reply tidb-cloud/migrate-incremental-data-from-mysql.md:76]: Done

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 15]
```diff
@@ -0,0 +1,267 @@
...
+
+This document describes how to migrate incremental data from MySQL-compatible databases to TiDB Cloud.
+
+## Before you begin
+
+Before you perform incremental data migration, you should have finished full data migration from MySQL to TiDB Cloud. For more information, see [Migrate Data from MySQL-Compatible Databases](/tidb-cloud/migrate-data-into-tidb.md).
+
+## Step 1. Deploy a DM Cluster
+
+The TiDB Cloud console does not provide incremental data migration feature yet. You need to deploy TiDB Data Migration (DM) manually to perform incremental migration to TiDB Cloud. For installation steps, see [Deploy a DM Cluster Using TiUP](https://docs.pingcap.com/tidb/stable/deploy-a-dm-cluster-using-tiup).
```
Comment (qiancai) [inline tidb-cloud/migrate-incremental-data-from-mysql.md:16]:
```suggestion
The TiDB Cloud console does not provide incremental data migration feature yet. You need to deploy [TiDB Data Migration](https://docs.pingcap.com/tidb/stable/dm-overview) (DM) manually to perform incremental migration to TiDB Cloud. For installation steps, see [Deploy a DM Cluster Using TiUP](https://docs.pingcap.com/tidb/stable/deploy-a-dm-cluster-using-tiup).
```

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 16]
```diff
@@ -0,0 +1,267 @@
...
+
+```shell
+[root@localhost ~]# tiup dmctl --master-addr ${advertise-addr} operate-source create dm-source1.yaml
+```
+
+The parameters used in the command above are described as follows:
+
+|Parameter              |Description    |
+|-                      |-              |
+|`--master-addr`        |The `{advertise-addr}` of any DM-master in the cluster where `dmctl` is to be connected, e.g.: 172.16.7.140:9261|
```
Comment (qiancai) [inline tidb-cloud/migrate-incremental-data-from-mysql.md:56]:
```suggestion
|`--master-addr`        |The `{advertise-addr}` of any DM-master node in the cluster where `dmctl` is to be connected. For example: 172.16.7.140:9261|
```

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 17]
```diff
@@ -0,0 +1,267 @@
...
+    events: ["drop database"]
+    action: Ignore
+
+##  ******** Ignore check items **********
+ignore-checking-items: ["table_schema"]
+```
+
+For detailed task configurations, see [DM Task Configurations](https://docs.pingcap.com/tidb/stable/task-configuration-file-full).
+
+To run a data migration task smoothly, DM triggers a precheck automatically at the start of the task and returns the check results. DM starts the migration only after the precheck is passed. To trigger a precheck manually, run the `check-task` command.
```
Comment (qiancai) [inline tidb-cloud/migrate-incremental-data-from-mysql.md:154]:
```suggestion
To run a data migration task smoothly, DM triggers a precheck automatically at the start of the task and returns the check results. DM starts the migration only after the precheck is passed. To trigger a precheck manually, run the `check-task` command:
```

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 18]
```diff
@@ -0,0 +1,267 @@
...
+
+```shell
+[root@localhost ~]# tiup dmctl --master-addr ${advertise-addr} start-task dm-task1.yaml
+```
+
+The parameters used in the command above are described as follows:
+
+|Parameter              |Description    |
+|-                      |-              |
+|`--master-addr`        |The {advertise-addr} of any DM-master in the cluster where `dmctl` is to be connected, e.g.: 172.16.7.140:9261|
```
Comment (qiancai) [inline tidb-cloud/migrate-incremental-data-from-mysql.md:183]:
```suggestion
|`--master-addr`        |The `{advertise-addr}` of any DM-master node in the cluster where `dmctl` is to be connected. For example: 172.16.7.140:9261|
```

**Review by qiancai — APPROVED** [review]
LGTM

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 19]
```diff
@@ -0,0 +1,267 @@
...
+summary: Learn how to migrate incremental data from MySQL to TiDB Cloud.
+---
+
+# Migrate Incremental Data from MySQL-Compatible Databases
+
+This document describes how to migrate incremental data from MySQL-compatible databases to TiDB Cloud.
+
+## Before you begin
+
+Before you perform incremental data migration, you should have finished full data migration from MySQL to TiDB Cloud. For more information, see [Migrate Data from MySQL-Compatible Databases](/tidb-cloud/migrate-data-into-tidb.md).
```
Comment (Rick-lee01) [inline tidb-cloud/migrate-incremental-data-from-mysql.md:12]:
```suggestion
Before you perform incremental data migration, you should have finished full data migration from MySQL-compatible databases to TiDB Cloud. For more information, see [Migrate Data from MySQL-Compatible Databases](/tidb-cloud/migrate-data-into-tidb.md).
```

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 20]
```diff
@@ -0,0 +1,267 @@
...
+source-id: "mysql-replica-01"
+
+# Configures whether DM-worker uses the global transaction identifier (GTID) to pull binlogs.
+# To enable this mode, the upstream MySQL must also enable GTID.
+# If the upstream MySQL service is configured to switch master between different nodes automatically, GTID mode is required.
+enable-gtid: true
+
+from:
+  host: "172.16.5.138"
+  user: "lzy"
```
Comment (Rick-lee01) [inline tidb-cloud/migrate-incremental-data-from-mysql.md:41]:
```suggestion
  host: "192.168.10.101"
  user: "user01"
```

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 21]
```diff
@@ -0,0 +1,267 @@
...
+Starting component `dmctl`: /root/.tiup/components/dmctl/v6.0.0/dmctl/dmctl /root/.tiup/components/dmctl/v6.0.0/dmctl/dmctl --master-addr 172.16.7.140:9261 operate-source create dm-source1.yaml
+{
+    "result": true,
+    "msg": "",
+    "sources": [
+        {
+            "result": true,
+            "msg": "",
+            "source": "mysql-replica-01",
+            "worker": "dm-172.16.7.154-9262"
```
Comment (Rick-lee01) [inline tidb-cloud/migrate-incremental-data-from-mysql.md:72]:
```suggestion
Starting component `dmctl`: /root/.tiup/components/dmctl/v6.0.0/dmctl/dmctl /root/.tiup/components/dmctl/v6.0.0/dmctl/dmctl --master-addr 192.168.11.110:9261 operate-source create dm-source1.yaml
{
    "result": true,
    "msg": "",
    "sources": [
        {
            "result": true,
            "msg": "",
            "source": "mysql-replica-01",
            "worker": "dm-192.168.11.120-9262"
```

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 22]
```diff
@@ -0,0 +1,267 @@
...
+
+```shell
+[root@localhost ~]# tiup dmctl --master-addr ${advertise-addr} check-task dm-task1.yaml
+```
+
+The following is an example output:
+
+```
+tiup is checking updates for component dmctl ...
+Starting component `dmctl`: /root/.tiup/components/dmctl/v6.0.0/dmctl/dmctl /root/.tiup/components/dmctl/v6.0.0/dmctl/dmctl --master-addr 172.16.7.140:9261 check-task dm-task1.yaml
```
Comment (Rick-lee01) [inline tidb-cloud/migrate-incremental-data-from-mysql.md:164]:
```suggestion
Starting component `dmctl`: /root/.tiup/components/dmctl/v6.0.0/dmctl/dmctl /root/.tiup/components/dmctl/v6.0.0/dmctl/dmctl --master-addr 192.168.11.110:9261 check-task dm-task1.yaml
```

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 23]
```diff
@@ -0,0 +1,267 @@
...
+Starting component `dmctl`: /root/.tiup/components/dmctl/v6.0.0/dmctl/dmctl /root/.tiup/components/dmctl/v6.0.0/dmctl/dmctl --master-addr 172.16.7.140:9261 start-task dm-task1.yaml
+{
+    "result": true,
+    "msg": "",
+    "sources": [
+        {
+           "result": true,
+            "msg": "",
+            "source": "mysql-replica-01",
+            "worker": "dm-172.16.7.154-9262"
```
Comment (Rick-lee01) [inline tidb-cloud/migrate-incremental-data-from-mysql.md:199]:
```suggestion
Starting component `dmctl`: /root/.tiup/components/dmctl/v6.0.0/dmctl/dmctl /root/.tiup/components/dmctl/v6.0.0/dmctl/dmctl --master-addr 192.168.11.110:9261 start-task dm-task1.yaml
{
    "result": true,
    "msg": "",
    "sources": [
        {
           "result": true,
            "msg": "",
            "source": "mysql-replica-01",
            "worker": "dm-192.168.11.120-9262"
```

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 24]
```diff
@@ -0,0 +1,267 @@
...
+{
+    "result": true,
+    "msg": "",
+    "sources": [
+        {
+            "result": true,
+            "msg": "",
+            "sourceStatus": {
+                "source": "mysql-replica-01",
+                "worker": "dm-172.16.7.154-9262",
```
Comment (Rick-lee01) [inline tidb-cloud/migrate-incremental-data-from-mysql.md:232]:
```suggestion
Starting component `dmctl`: /root/.tiup/components/dmctl/v6.0.0/dmctl/dmctl /root/.tiup/components/dmctl/v6.0.0/dmctl/dmctl --master-addr 192.168.11.110:9261 query-status test-task1
{
    "result": true,
    "msg": "",
    "sources": [
        {
            "result": true,
            "msg": "",
            "sourceStatus": {
                "source": "mysql-replica-01",
                "worker": "dm-192.168.11.120-9262",
```

**Review by Rick-lee01 — COMMENTED** [review]
Please replace the real IP and username.

File: tidb-cloud/migrate-incremental-data-from-mysql.md [hunk 25]
```diff
@@ -0,0 +1,267 @@
+---
+title: Migrate Incremental Data from MySQL-Compatible Databases
+summary: Learn how to migrate incremental data from MySQL to TiDB Cloud.
```
Comment (hfxsd) [inline tidb-cloud/migrate-incremental-data-from-mysql.md:3]:
```suggestion
summary: Learn how to migrate incremental data from MySQL-compatible databases to TiDB Cloud.
```

**Review by Rick-lee01 — APPROVED** [review]
LGTM

**Pull request merged**